BlueprintUI/Sources/CrossLayoutCaching/Cache/HostingViewContext.swift

```diff
@@ -0,0 +1,85 @@
+import Foundation
+import UIKit
+
+/// Environment-associated storage used to cache types used across layout passes (eg, size calculations).
+/// The storage itself is type-agnostic, requiring only that its keys and values conform to the `CrossLayoutCacheKey` protocol
+/// Caches are responsible for managing their own lifetimes and eviction strategies.
+@_spi(HostingViewContext) public final class HostingViewContext: Sendable, CustomDebugStringConvertible {
+
+    // Optional name to distinguish between instances for debugging purposes.
+    public var name: String? = nil
+    fileprivate var storage: [ObjectIdentifier: Any] = [:]
+
+    init() {
+        NotificationCenter.default.addObserver(
+            forName: UIApplication.didReceiveMemoryWarningNotification,
+            object: nil,
+            queue: .main
+        ) { [weak self] _ in
+            self?.storage.removeAll()
+        }
+    }
+
+    public subscript<KeyType>(key: KeyType.Type) -> KeyType.Value where KeyType: CrossLayoutCacheKey {
+        get {
+            storage[ObjectIdentifier(key), default: KeyType.emptyValue] as! KeyType.Value
+        }
+        set {
+            storage[ObjectIdentifier(key)] = newValue
+        }
+    }
+
+    public var debugDescription: String {
+        let debugName = if let name {
+            "HostingViewContext (\(name))"
+        } else {
+            "HostingViewContext"
+        }
+        return "\(debugName): \(storage.count) entries"
+    }
+
+}
+
+extension Environment {
+
+    struct HostingViewContextKey: InternalEnvironmentKey {
```

Comment:
Plucking this question out of an old thread:

Can we avoid the need for `InternalEnvironmentKey` by making this key use a reference comparison of `HostingViewContext`, or hardcoded `true`?